src/lib/network/network_autoyast.rb

```diff
@@ -281,20 +281,24 @@ def assign_udevs_to_devs(udev_rules)
         key.downcase!
 
         # find item which matches to the given rule definition
-        item, matching_item = LanItems.Items.find do |_, i|
-          i["hwinfo"] &&
-            (i["hwinfo"]["busid"].downcase == key || i["hwinfo"]["mac"].downcase == key)
+        item_id, matching_item = LanItems.Items.find do |_, i|
+          next unless i["hwinfo"]
+          busid = i["hwinfo"]["busid"]
+          parent_busid = i["hwinfo"]["parent_busid"] || busid
```

Comment:
this is eq to ```parent_busid = i["hwinfo"]["parent_busid"] || i["hwinfo"]["busid"]``` so you don't need to assign and compare ```busid``` bellow at all

Reply:
If it have a parent busid it will be the parent_bus_id not the busid. so the busid need to be compared.

Reply:
Basically:

1) No parent_busid, it becomes the busid, so it is like a duplicate matching that does not hurt

`["mac", "busid","busid"]`

2) parent_busid is present, it becomes the parent

`["mac", "parent_busid","busid"]`

Reply:
i wasn't that clear how i should be.

what i wanted to say that it doesn't hurt but is useless ... in the end you compare either ```busid``` or ```parent_busid```. if i write differently what I wrote above then you can define ```busid_to_use_in_comparison = i["hwinfo"]["parent_busid"] || i["hwinfo"]["busid"]``` and then
````[busid_to_use_in_comparison, mac].any? { |v| v.downcase == key }``

Reply:
But we want to compare both

Reply:
Ok, seems I'm getting lost.

Do they want to match device either according to busid or its parent busid? That's not a bug but 
feature and should be implemented as such by extending the AY rule to accept not only BUSID and MAC but even parent BUSID. 

Moreover, when matched by parent id I don't think it will generate udev rule as expected.

Reply:
Have you seen the trello card then? I should update the description :)

Reply:
> Ok, seems I'm getting lost.
> 
> Do they want to match device either according to busid or its parent busid? That's not a bug but
> feature and should be implemented as such by extending the AY rule to accept not only BUSID and MAC but even parent BUSID.
> 
> Moreover, when matched by parent id I don't think it will generate udev rule as expected.

We will replace the udev rule by the one defined in the profile. I already tested it by hand and the user also verified the patch

Reply:
i just want to understand it as much as possible ... bcs i dislike it ;-)

pls add some explaining comment or reference. Mainly expressing that virtio is sometime identified by parentid and sometime by virtio id